devito/petsc/logging.py

```diff
@@ -142,20 +167,41 @@ def section(self):
 
     @property
     def summary_key(self):
-        return (self.section, self.sobjs['options_prefix'])
+        # If users provide an options prefix, use it in the summary;
+        # otherwise, use the default one generated by Devito
+        user_prefix = self.inject_solve.expr.rhs.user_prefix
+        if user_prefix is None:
+            user_prefix = self.formatted_prefix
+        return (self.section, user_prefix)
 
     def __getattr__(self, attr):
-        if attr in self.logobjs.keys():
-            return getattr(self.value._obj, self.logobjs[attr].name)
-        raise AttributeError(f"{attr} not found in PETSc return variables")
+        if attr not in self.petsc_option_mapper:
+            raise AttributeError(f"{attr} not found in PETSc return variables")
+
+        # Maps the petsc_option to its generated variable name e.g {'its': its0}
+        obj_mapper = self.petsc_option_mapper[attr]
+
+        # Decode the value if it is a bytes object
+        def decode_if_bytes(val):
+            return str(os.fsdecode(val)) if isinstance(val, bytes) else val
+        get_val = lambda v: decode_if_bytes(getattr(self.value._obj, v.name))
```

Comment:
This seems slightly contrived? Can this lambda not just be part of the function one line above?